### PR TITLE
334 stripe payment system

### LINKED ISSUE #334: FeedAHelp: Investigate and implement Stripe
Stripe: Stripe is a widely used payment gateway that supports online payments in Europe and many other regions. It offers a simple integration process, supports multiple currencies, and provides robust developer tools. Stripe is known for its flexibility and the ability to handle subscription billing and e-commerce transactions.

### PR DESCRIPTION
**Pull request by MNSM92**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #334

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: apps/feedahelp/src/components/ui/features/payments/Payment.tsx [hunk 1]
```diff
@@ -45,6 +47,10 @@ const Payment = () => {
               />
             </PayPalScriptProvider>
           </TabPanel>
+          <TabPanel value="3">
+            <Page/>
```
Comment (TanvirAlam) [inline apps/feedahelp/src/components/ui/features/payments/Payment.tsx]: rename to `<StripScriptProvider />`

File: apps/feedahelp/src/components/ui/features/payments/Payment.tsx [hunk 2]
```diff
@@ -45,6 +47,10 @@ const Payment = () => {
...
+
```
Comment (TanvirAlam) [inline apps/feedahelp/src/components/ui/features/payments/Payment.tsx]: remove empty line

File: apps/feedahelp/src/components/ui/features/payments/Stripe/CheckoutForm.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+"use client";
+import { useState } from "react";
+import getStripe from "./GetStripe";
+import axios from "axios";
+import { useSession } from "next-auth/react";
+
+const baseBtnStyle =
+  "bg-slate-100 hover:bg-slate-200 text-black px-6 py-2 rounded-md capitalize font-bold mt-1";
```
Comment (TanvirAlam) [inline apps/feedahelp/src/components/ui/features/payments/Stripe/CheckoutForm.tsx]: move this to `styles/global.css`

File: apps/feedahelp/src/components/ui/features/payments/Stripe/CheckoutForm.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+export default function Page() {
+  const { data: session } = useSession();
+  const [value, setValue] = useState("10")
+  const email= session?.user?.email ?? undefined
+  const handleCreateCheckoutSession = async () => {
+    try {
+      const response = await axios.post(
+        `http://localhost:4000/api/v1/stripe/create`, {value, email, currency: 'usd'}
```
Comment (TanvirAlam) [inline apps/feedahelp/src/components/ui/features/payments/Stripe/CheckoutForm.tsx]: get this from `.env`, we already have base URL use it

File: apps/server/routes/stripe.router.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import express, { Router } from "express";
+import handler from '../controllers/stripe/stripe.controller';
+// import webhookHandler from '../controllers/stripe/webhooks.controller';
```
Comment (TanvirAlam) [inline apps/server/routes/stripe.router.ts]: stop commeting stuff, if not needed delete them

File: apps/server/routes/stripe.router.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+
+export const stripeRouter: Router = express.Router();
+
+stripeRouter.post('/create', handler)
+// stripeRouter.post('/webhook', webhookHandler)
```
Comment (TanvirAlam) [inline apps/server/routes/stripe.router.ts]: same here

File: apps/server/controllers/stripe/stripe.controller.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+              currency: currency,
+              product_data: {
+                name: 'One Time Donation. Thanks for Your Big Hand.'
+              },
+              unit_amount: +value * 100
+            },
+            quantity: 1
+          }
+        ],
+        success_url: `http://localhost:3010/`,
```
Comment (TanvirAlam) [inline apps/server/controllers/stripe/stripe.controller.ts]: you are hard coding? what will happen in production?

File: apps/server/controllers/stripe/stripe.controller.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        cancel_url: `http://localhost:3010/canceled`
```
Comment (TanvirAlam) [inline apps/server/controllers/stripe/stripe.controller.ts]: same here?

**Review by TanvirAlam — CHANGES_REQUESTED** [review]
Nice work, do small chages het back @MNSM92

File: apps/feedahelp/src/components/ui/features/payments/Payment.tsx (hunk 1 above)
Comment (MNSM92) [reply apps/feedahelp/src/components/ui/features/payments/Payment.tsx]: Done

**Review by TanvirAlam — APPROVED** [review]
LGTM

**Pull request merged**